src/shared/install.c

```diff
@@ -334,6 +334,10 @@ void unit_file_dump_changes(int r, const char *verb, const UnitFileChange *chang
                                 log_info("Unit %s is an alias to a unit that is not present, ignoring.",
                                          changes[i].path);
                         break;
+                case UNIT_FILE_WARNING:
+                        if (!quiet)
+                                log_warning("Installed unit %s into a non-existent target %s.", changes[i].source, changes[i].path);
```

Comment:
this doesn't have to be a target btw, could be any type of unit. hence `s/target/unit/`